### PR TITLE
[SEDONA-610] Add ST_IsValidTrajectory

### PR DESCRIPTION
**Pull request by furqaankhan**
## Did you read the Contributor Guide?

- Yes, I have read the [Contributor Rules](https://sedona.apache.org/latest-snapshot/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest-snapshot/community/develop/)

## Is this PR related to a JIRA ticket?

- Yes, the URL of the associated JIRA ticket is https://issues.apache.org/jira/browse/SEDONA-610. The PR name follows the format `[SEDONA-XXX] my subject`.

## What changes were proposed in this PR?

- Add ST_IsValidTrajectory

## How was this patch tested?

- add new unit tests

## Did this PR include necessary documentation updates?

- Yes, I have updated the documentation.

### CODE REVIEW

**Review by jiayuasu — CHANGES_REQUESTED** [review]
please remove `println`

**Comment by furqaankhan** [comment]
It is there to provide the user with information. [Reference.](https://github.com/postgis/postgis/blob/44f71532565df5492a5c0a6e8721c097356eb627/liblwgeom/lwline.c#L454). Let me know if you still want it removed.

**Comment by jiayuasu** [comment]
This is not the correct way to provide info. We should use log4j for this purpose. @Kontinuation maybe it is time to add log4j dependencies to the `sedona-common` module?

**Comment by furqaankhan** [comment]
I used print statements just because the logger wasn't available to me.

File: docs/api/sql/Function.md [hunk 1]
```diff
@@ -2201,6 +2201,28 @@ gid  |                  validity_info
...
+## ST_IsValidTrajectory
+
+Introduction: This function checks if a geometry is a valid trajectory representation. For a trajectory to be considered valid, it must be a LineString that includes measure (M) values. The key requirement is that the M values increase from one vertex to the next as you move along the line.
+
+Format: `ST_IsValidTrajectory(geom: Geometry)`
+
+Since: `v1.6.1`
+
+SQL Example:
+
```
Comment (jiayuasu) [inline docs/api/sql/Function.md:2213]: Please add an example that returns false

**Review by jiayuasu — CHANGES_REQUESTED** [review]

**Comment by jiayuasu** [comment]
@furqaankhan please still remove the println command.

**Review by jiayuasu — APPROVED** [review]

**Review by jiayuasu — APPROVED** [review]

**Pull request merged**